BraveShared/Analytics/UserReferralProgram.swift

```diff
@@ -180,6 +187,19 @@ public class UserReferralProgram {
         return nil
     }
     
+    /// Passing string, attempts to derive a ref code from it.
+    /// Uses very strict matching.
+    /// Returns the sanitized code, or nil if no code was found
+    public class func sanitize(input: String?) -> String? {
+        guard var input = input, input.hasPrefix(self.clipboardPrefix) else { return nil }
+        
+        input.removeFirst(self.clipboardPrefix.count + 1)
+        // Add any other potential validation here, e.g. validating the actual ref code string
```

Comment:
https://docs.google.com/document/d/15oSBC-57OpLO3pZK_XjHa1SwIjnjuxA8Wi74K5ZaxPc/edit?disco=AAAAJWaIrNI - `[0-9A-Z]{6}` should work

Reply:
The ref code is guaranteed to be 6 upper-case, 0-9 codes?

Reply:
Refcode must be ABC123 format, 3 uppercase alphabetic and 3 decimal digits.

Reply:
@jumde Please finalize this on the spec first.

Reply:
Updated regex per spec.